Make space dazzle #96

s-i-s opened the pull request:
Closes #96 
Updated Space scene:
- refactored code to be structurally similar to Knots, Bubbles, Cubes
- Space scene now has:
    (1) a starry sky that changes color according to primary emotion (same as before)
    (2) a starfield particle system that changes color according to primary personality
    (3) scattered, rotating planets of varying textures and sizes

- Not currently rendered: code for a circle of planets, each with their own orbit of small spheres. Could be used in future for avatars in space scene when we debug the Avatar/roster code

- Possible next steps:
   - bring in / tweak our own particle system so we can animate more effectively
   - have movement of planets and/or starfield respond to indico output

s-i-s committed:
changes in Room to add updated Space scene for testing

s-i-s committed:
successfully refactored Scene to initialize avatars, light, and camera

s-i-s committed:
implemented dust-like particle system that fades from fuchsia to blue

s-i-s committed:
star particle system looks like actual stars and change colors in response to personality data

s-i-s committed:
tweaked the star particle system and planets. space is dazzling

s-i-s committed:
added functionality to rotate planets, create circular orbits

s-i-s committed:
added functionality to make a circle of planet avatars each with individual orbit. not yet wired up to actual avatar component

s-i-s committed:
cleaned up code for Space Scene

s-i-s committed:
resolving merge conflicts with master

thefishter commented:
this is awesome and so organized. a few things to check in about:

1. in `space.jsx`, why does it say `export default class Scene extends Component` - shouldn't that be `Space`? especially because in `Room.jsx` you have`import Space from './Space.jsx'`

2. in the interest of drying out code, `initScene1()` and `initScene2()` in `Space.jsx` might become one function that takes an argument (1 or 2) and utilizes a switch statement for the differing portion of code

3. let's take out large commented chunks (e.g. `space.js`) - they're already saved in the branch!

s-i-s committed:
Update Space.jsx

combined initScene1 and initScene2 into 1 function

s-i-s committed:
Update Space.jsx

corrected typo -- export default class Space (not Scene)

s-i-s committed:
Update space.js

Deleted 'code in progress' (finding an alternate way to animate the particle system) -- can refer to past commits if necessary

s-i-s commented:
Fish - done! 

(except I used if statements instead of a switch statement because that's what I'm used to -- feel free to change if we are trying to keep it stylistically consistent with some other parts of the app)
